lib/transfer.c

```diff
@@ -1222,6 +1222,13 @@ CURLcode Curl_readwrite(struct connectdata *conn,
         infof(data, "Done waiting for 100-continue");
       }
     }
+
+#ifdef ENABLE_QUIC
+    if(conn->transport == TRNSPRT_QUIC)
+      result = Curl_quic_idle(data);
+    if(result)
+      return result;
```

Comment:
The contract of this API isn't immediately clear to me, do we want library housekeeping functions be able to abort the transfer here? Also, other subsystem housekeeping, like removing expired cookies etc, is pushed to the respective subsystem to perform at leisure, if we introduce something like this maybe we should design something which can offload more generically?

Reply:
The thing that sort of bothers me is that it feels like a layering violation of `Curl_handler`, in that before adding this we should perhaps see if this is a problem generically faced (but not currently addressed) which we can solve in a bigger picture context.

This might be completely wrong but I think the discussion is worth having before special casing.

Reply:
If we do not have this kind of problem before QUIC is introduced, we do not need to make this generic.
QUIC itself is quite special in a sense that it is a proper transport protocol, the same layer of TCP.
Either way, without this kind of facility, curl is unable to handle QUIC properly.